FlowCrypt/Controllers/Inbox/InboxViewController+TableView.swift

```diff
@@ -417,7 +417,7 @@ extension InboxViewController {
         case .moveToTrash, .permanentlyDelete:
             removeMessage(at: indexToUpdate)
         case .archive, .moveToInbox:
-            if path.isEmpty { // no need to remove in 'All Mail' folder
+            if path.isEmpty || path == "SENT" { // no need to remove in 'All Mail' folder
```

Comment:
it looks like we should remove messages only on `Inbox` screen, as moving to inbox or archiving doesn't remove messages from other folders.

For example, go to some label folder, archive message there - it should still stay in messages list of this label, but currently it's removed

Reply:
Yep! Correct